src/dto/sampling-train.dto.ts

```diff
@@ -0,0 +1,34 @@
+export class SamplingTrainBaseDTO {
+  componentId?: string;
```

Comment:
XML has nillable="false"

So component_id column is nullable: true in sampling train entity but what the DTO is referring to is actually coming from the column component_identifier in the component table after joining over on component_id. It's just confusing because the name is the same as primary column in the sampling train entity